PocketKit/Sources/PocketKit/Premium/Model/PremiumSubscriptionStore.swift

```diff
@@ -5,15 +5,117 @@
 import SharedPocketKit
 import StoreKit
 
+/// Subscription store error(s)
+enum SubscriptionStoreError: Error {
+    case unverifiedPurchase
+}
+
 /// A type that handles premium subscriptions purchases from the App Store
 final class PremiumSubscriptionStore: ObservableObject {
     @Published private(set) var subscriptions: [PremiumSubscription] = []
+    private var user: User
+    /// Will listen for transaction updates while the app is running
+    private var transactionListener: Task<Void, Error>?
+
+    // TODO: determine what we actually need to store here
+    /// For premium users, this property contains details of the purchased subscription
+    @Published private(set) var purchasedSubscription: PremiumSubscription?
+
+    init(user: User) {
+        self.user = user
+        transactionListener = makeTransactionListener()
 
-    init() async throws {
-        try await requestSubscriptions()
+        Task {
+            do {
+                try await requestSubscriptions()
+            } catch {
+                // TODO: use logger here
+                print(error)
+            }
+        }
     }
-    @MainActor
```

Comment:
curious, why remove `@MainActor`

Reply:
Initially this call was directly made at `PremiumUpgradeView` instantiation. Now, we instantiate the store at app launch and pre-fetch products. This method is also called by `PremiumUpgradeViewModel` when the corresponding view is instantiated, but being that view model marked as `@Mainactor` will ensure that UI updates will still happen on the main thread.